src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -33,12 +33,17 @@ public function __construct(
 
     public function getMetadata(string $name): LiveComponentMetadata
     {
+        static $liveComponentMetadata;
```

Comment:
Maybe a private typed property ?

Reply:
I could change it to

    /** @var LiveComponentMetadata[] */
    private static array $liveComponentMetadata = [];

    ...

    public function getMetadata(string $name): LiveComponentMetadata
    {
        if (isset(self::$liveComponentMetadata[$name])) {
            return self::$liveComponentMetadata[$name];
        }
        
        $componentMetadata = $this->componentFactory->metadataFor($name);

        $reflectionClass = new \ReflectionClass($componentMetadata->getClass());
        $livePropsMetadata = $this->createPropMetadatas($reflectionClass);

        return self::$liveComponentMetadata[$name] = new LiveComponentMetadata($componentMetadata, $livePropsMetadata);
    }

Reply:
Yup - do this. But don't make the property static. In practice, there will only be one `LiveComponentMetadataFactory` service, so static vs non-static won't make any real difference. But we should not *enforce* that there can only be one instance of this object by using static.